### ISSUE TITLE
Add PDF plugin

### ISSUE THREAD

**Issue by sspencerwire** [rocky-linux/docs.rockylinux.org#23]
I've received a request from one of the contributors to add a plugin to mkdocs to allow for generating a PDF. The PDF plugin is `mkdocs-with-pdf` and he has tested it locally to generate specific PDF's. This, I think, is a good suggestion, but it may need to be reviewed by others to see what negative impact it might have. Thanks for your consideration! Steve

**Comment by sspencerwire** [comment]
@hbjydev Could you review this issue and see if it has merit?  Thanks, Steve

**Comment by hbjydev** [comment]
I'm... Not too sure why we _couldn't_, unless it relies on something server-side, or adds a ton of build time.

I'll have a play with it and get back to you but we probably could.

**Comment by sspencerwire** [comment]
Let me know either way and thanks for taking a look!

**Comment by sspencerwire** [comment]
@hbjydev Remember this issue? The contributor who originally asked for this feature is asking again. I think you were going to do some tests to see what sort of load this added to the build time. I'm not sure where that landed or if you got side-tracked with other more important issues. So I'm just raising the flag again. Let me know, and Thanks!  Steve

**Comment by hbjydev** [comment]
@sspencerwire Admittedly, this has taken me a long time to get around to. But I'll look into implementing this later on today :)

**Comment by alemorvan** [comment]
You can create a PDF with this example:

```
---
site_name: 'Documentation'
docs_dir: '../rocky-documentation/docs/books/admin_guide/'

nav:
  - 00-toc.md
  - 01-presentation.md
  - 03-commands.md
  - 04-advanced-commands.md
  - 05-vi.md
  - 06-users.md
  - 07-file-systems.md
  - 08-process.md
  - 09-backups.md
  - 10-boot.md
  - 11-tasks.md
  - 12-network.md
  - 13-softwares.md

markdown_extensions:
  - abbr
  - attr_list
  - admonition
  - toc:
      permalink: true
  - pymdownx.emoji:
      emoji_index: !!python/name:materialx.emoji.twemoji
      emoji_generator: !!python/name:materialx.emoji.to_svg
  - pymdownx.highlight
  - pymdownx.superfences
  - pymdownx.keys
  - pymdownx.tabbed
  - pymdownx.details
  - pymdownx.tasklist
  - footnotes
  - def_list
  - meta

plugins:
  - awesome-pages
  - with-pdf:
      #cover_title:
      cover_subtitle: Admin Guide
      cover_logo: theme/assets/images/logo.png
      back_cover: true
      toc_level: 2
      output_path: ../document.pdf

copyright: Copyright &copy; 2021 The Rocky Enterprise Software Foundation
```

Please change the location of your `docs_dir` then launch `mkdocs build -f mkdocs-pdf.yml` .

- You may need to install weasyprint: https://doc.courtbouillon.org/weasyprint/latest/first_steps.html#linux
- See requirements here: https://github.com/orzih/mkdocs-with-pdf#requirements

The build took 71 sec on my VM for a 148 pages PDF file.

**Comment by alemorvan** [comment]
Fix by #40

**Comment by alemorvan** [comment]
@ressonix please close this PR (fix by #40)
Thx.
